src/common/lib/types/protocolmessage.ts

```diff
@@ -83,7 +83,8 @@ export function fromDeserialized(
   const error = deserialized.error;
   if (error) deserialized.error = ErrorInfo.fromValues(error as ErrorInfo);
   const messages = deserialized.messages as Message[];
-  if (messages) for (let i = 0; i < messages.length; i++) messages[i] = messageFromValues(messages[i]);
+  if (messages)
+    for (let i = 0; i < messages.length; i++) messages[i] = messageFromValues(messages[i], { stringifyAction: true });
```

Comment:
(I don't want to comment too much on the coding style of a project I'm not a primary contributor to any more, but is this really the preferred coding style for ably-js these days? No braces for a multiline `if`, and the body of a `for` on the same line as the condition? Seems an.. unusual choice)

Reply:
I can't really comment on that either as this is my first contribution , but it seems to be the norm throughout. I usually try and follow the coding style of the repo, but in this case I think I'll make the change to something more structured.